src/deluge/gui/views/timeline_view.cpp

```diff
@@ -146,24 +151,36 @@ ActionResult TimelineView::horizontalEncoderAction(int32_t offset) {
 
 		if (isUIModeActiveExclusively(UI_MODE_HOLDING_HORIZONTAL_ENCODER_BUTTON)) {
 			int32_t oldXZoom = currentSong->xZoom[navSysId];
-
 			int32_t zoomMagnitude = -offset;
+			uint32_t newZoom = zoomMagnitude == -1 ? oldXZoom >> 1 : oldXZoom << 1;
+			uint32_t clipLengthMaxZoom = getMaxZoom();
 
-			// Constrain to zoom limits
-			if (zoomMagnitude == -1) {
-				if (currentSong->xZoom[navSysId] <= 3) {
+			// When this timer is set, we may need to wait a bit before allowing zooming
+			if (timeHorizontalZoomLastHitClipMax != 0) {
+				bool isWithinZoomDelay =
+				    AudioEngine::audioSampleTimer - timeHorizontalZoomLastHitClipMax < kShortPressTime;
```

Comment:
this should use `infinite_a_lt_b` from utils since it handles wrapping, otherwise we'll get stuck if there's a particularly unfortunate wrap.

Reply:
Like this?

```suggestion
				    util::infinite_a_lt_b(AudioEngine::audioSampleTimer - timeHorizontalZoomLastHitClipMax, kShortPressTime)
```

I'm not sure I really understand the issue. Also, the method of doing this without `infinite_a_lt_b` is definitely used in a bunch of the existing code. I pretty much ripped this way of measuring a delay from a number of other examples that do the same thing.

Reply:
yeah that function didn't exist until a couple months ago so a bunch of existing code doesn't use it. that code is all buggy it just doesn't matter often in practice since the wrap interval on the sample timer is so long.